### DISCUSSION TITLE
Why is the versioning of this application so weird?

### DISCUSSION THREAD

**Discussion by AnonymousWP** [celzero/rethink-app#1646] [Q&A] [answered]
As the title states: why is the versioning so odd/weird? I didn't see this kind of versioning before, where you use semantic versioning + a letter. Why not keeping it 'normal' semantic versioning? Like 0.5.5 -> 0.5.6 -> 0.5.7? Of course, it's not a real problem, but it reads easier and I don't understand why you'd take another approach if most open-source projects use semantic versioning.

Otherwise: I've been using this app again since a few months after being absent (due to critical bugs, which made it unusable), but now I'm using it it's great. 😄

**Comment by ignoramous** [comment]
Version names (git tags) are merely *build* metadata for this project. *Version Code* is where all the action is at.

I hope it doesn't bother you too much. It is going to remain the same or perhaps get even more weirder 🙃

**Reply by AnonymousWP** [reply]
I still don't really get it. 90% of the other repositories, including very large ones (just look at any big corp's repositories and its releases) use semantic versioning which makes a lot of sense for the user. I just don't see why you *wouldn't* follow the 'normal' scheme of version numbers. Sure, they're build metadata, but your version code is nowhere visible (not in the app at least, which should be the least thing imo).

In the end it doesn't matter much, it's just a bit baffling for me as it doesn't make sense to me (mainly because repos follow best practices), but this doesn't seem to be the case for this repo.

**Reply by ignoramous** [reply]
You've a point, but semantic versioning isn't really meant for "consumers" (end-user) but for other developers. As above, *version code* is what matters since that's the basis of Android's package management, not the *version name* (which, in Android, is relegated to build-metadata that's visible to the end-user).

**Reply by AnonymousWP** [reply]
Why isn't it? If it wouldn't be, then many applications that are actively used by all kinds of users (even non tech-savy users) wouldn't be using semantic versioning either, and they are (except those of big corps). ReThink isn't really for the average user though. Why would it be *more* important for other developers than for us users? The versioning makes even less sense as:

It's unclear if something is a 'major' version: new features are even added when the number isn't bumped (it's just a letter change). I don't think I'll change your opinion, but for me and many others it helps choosing whether I should wait with updating (I generally wait a few versions before a new major update is out).

**Reply by ignoramous** [reply]
You make valid points.

>  Why would it be more important for other developers than for us users?

Because versioning affects how modules, dependencies, libraries, frameworks, SDK, APIs/ABIs, services etc (things developers worry about) are consumed by downstream code. Semantic versioning is therefore (among other things) a signal whether a particular version upgrade will be backwards compatible or not, and if it may break, how so (release docs generated with version diffs).

> ReThink isn't really for the average user though

We certainly want to pivot in that direction, though, the jury is out (:

> but for me and many others it helps choosing whether I should wait with updating

True. We try to be consistent though. everytime we do a major change, we bump up the *minor* version (:

**Reply by AnonymousWP** [reply]
Thanks for trying to understand. So if I make valid points, there's no way/chance that you'll be guys stepping away from this very abnormal way of versioning?😄

**Reply by ignoramous** [reply]
I understand that it is 'abnormal' but we are not alone in our folly (nor would we be alone in breaking away from `0`ver).[^0]

[^0]: https://0ver.org/

**Comment by ignoramous** [comment, marked as answer]
Someday...

- #1871

**Reply by AnonymousWP** [reply]
Really hoping that.
